xarray/conventions.py

```diff
@@ -476,18 +477,41 @@ def stackable(dim: Hashable) -> bool:
         if decode_coords == "all":
             for attr_name in CF_RELATED_DATA:
                 if attr_name in var_attrs:
-                    attr_val = var_attrs[attr_name]
-                    if attr_name not in CF_RELATED_DATA_NEEDS_PARSING:
-                        var_names = attr_val.split()
-                    else:
-                        roles_and_names = [
-                            role_or_name
-                            for part in attr_val.split(":")
-                            for role_or_name in part.split()
-                        ]
-                        if len(roles_and_names) % 2 == 1:
-                            emit_user_level_warning(f"Attribute {attr_name} malformed")
-                        var_names = roles_and_names[1::2]
+                    # fixes stray colon
+                    attr_val = var_attrs[attr_name].replace(" :", ":")
+                    var_names = attr_val.split()
+                    # if grid_mapping is a single string, do not enter here
+                    if (
+                        attr_name in CF_RELATED_DATA_NEEDS_PARSING
+                        and len(var_names) > 1
+                    ):
+                        # map the keys to list of strings
+                        roles_and_names = defaultdict(list)
+                        key = None
+                        for vname in var_names:
+                            if ":" in vname:
+                                key = vname.strip(":")
+                            else:
+                                if key is None:
```

Comment:
I'm confused here. Is this handling `A: b c d`?

Reply:
Yes, `A: b c d E: f g` will return '{"A": ["b", "c", "d"], "E": ["f", "g"]}'.